field/track.go

```diff
@@ -0,0 +1,404 @@
+package field
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"regexp/syntax"
+	"strings"
+	"time"
+
+	"github.com/moov-io/iso8583/utils"
+)
+
+var _ Field = (*Track)(nil)
+
+type Track struct {
+	spec *Spec  `xml:"-" json:"-"`
+	data *Track `xml:"-" json:"-"`
+
+	// Available for Track 1
+	FixedLength bool `xml:"-" json:"-"`
+
+	// Available versions 1,2,3
+	Version TrackVersion `xml:"Version,omitempty" json:"version,omitempty"`
+
+	// Available for Track 1, Track 3
+	FormatCode string `xml:"FormatCode,omitempty" json:"format_code,omitempty"`
+
+	// Available for Track 1, Track 2, Track 3
+	PrimaryAccountNumber string `xml:"PrimaryAccountNumber,omitempty" json:"primary_account_number,omitempty"`
+
+	// Available for Track 1
+	Name string `xml:"Name,omitempty" json:"name,omitempty"`
+
+	// Available for Track 1, Track 2
+	ExpirationDate *time.Time `xml:"ExpirationDate,omitempty" json:"expiration_date,omitempty"`
+
+	// Available for Track 1, Track 2
+	ServiceCode string `xml:"ServiceCode,omitempty" json:"service_code,omitempty"`
+
+	// Available for Track 1, Track 2, Track 3
+	//  If track 3, the field describe security data + additional data
+	DiscretionaryData string `xml:"DiscretionaryData,omitempty" json:"discretionary_data,omitempty"`
+}
+
+type TrackVersion int
+
+func (v TrackVersion) Valid() bool {
+	if v == VersionFirst || v == VersionSecond || v == VersionThird {
+		return true
+	}
+	return false
+}
+
+const (
+	VersionFirst  = 1
+	VersionSecond = 2
+	VersionThird  = 3
+)
+
+var (
+	expiryDateFormat   = "0601"
+	trackFirstRegex    = regexp.MustCompile(`^([A-Z]{1})([0-9]{1,19})\^([^\^]{2,26})\^([0-9]{4}|\^)([0-9]{3}|\^)([^\?]+)$`)
+	trackSecondRegex   = regexp.MustCompile(`^([0-9]{1,19})\=([0-9]{4}|\=)([0-9]{3}|\=)([^\?]+)$`)
+	trackThirdRegex    = regexp.MustCompile(`^([0-9]{2})([0-9]{1,19})\=([^\?]+)$`)
+	trackFirstPattern  = `^([A-Z]{1})([0-9]{1,19})\^([^\^]{2,26})\^([0-9]{4}|\^)([0-9]{3}|\^)([^\?]+)$`
+	trackSecondPattern = `^([0-9]{1,19})\=([0-9]{4}|\=)([0-9]{3}|\=)([^\?]+)$`
+	trackThirdPattern  = `^([0-9]{2})([0-9]{1,19})\=([^\?]+)$`
+)
+
+func NewTrack(spec *Spec, version TrackVersion) (*Track, error) {
+	if !version.Valid() {
+		return nil, errors.New("invalid track version")
+	}
+	return &Track{
+		spec:    spec,
+		Version: version,
+	}, nil
+}
+
+func NewTrack1Value(val []byte, version TrackVersion, isFixed bool) (*Track, error) {
```

Comment:
nit: I wasn't sure what `isFixed` meant at first
 ```suggestion
func NewTrack1Value(val []byte, version TrackVersion, fixedLength bool) (*Track, error) {
```